### PR TITLE
feat(mutationmodal): added mutation modal for payout requests

### LINKED ISSUE #293: [Bug]: Transaction table cannot handle uncommon mutation types
### What browsers are you seeing the problem on?

Firefox

### Is this happening on the SudoSOS point of sale or on the SudoSOS dashboard?

Dashboard

### Current behaviour

Transaction table claims all uncommon transfer types such as invoices etc are made by "You". You cannot open the modal for the invoices, and the modal for payout requests show nothing. 

### Desired behaviour

It should work roughly the same as top ups and transactions

### Steps to reproduce

Go to Samuel's account in the test database, as well as "Facturen account" on the production database

### Other information

_No response_

### PR DESCRIPTION
**Pull request by CodeNamedRobin**
# Description
Adds a mutation modal specifically for payout requests.

## Related issues/external references
Closes #293 

## Types of changes

- Bug fix _(non-breaking change which fixes an issue)_
- New feature _(non-breaking change which adds functionality)_

### CODE REVIEW

File: apps/dashboard/src/components/mutations/mutationmodal/ModalDetailPayoutRequest.vue [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+const props = defineProps({
+  payoutRequest: {
+    type: Object as () => TransferResponse,
+    required: true,
+  }
+});
+
+onMounted(async () => {
+  console.warn(props.payoutRequest);
```
Comment (SuperVK) [inline apps/dashboard/src/components/mutations/mutationmodal/ModalDetailPayoutRequest.vue]: beep boop

**Comment by SuperVK** [comment]
Might be nice to immediatly also fix the other issues in #293

File: apps/dashboard/src/components/mutations/mutationmodal/ModalDetailPayoutRequest.vue (hunk 1 above)
Comment (CodeNamedRobin) [reply apps/dashboard/src/components/mutations/mutationmodal/ModalDetailPayoutRequest.vue]: Haha woops

**Comment by CodeNamedRobin** [comment]
> Might be nice to immediatly also fix the other issues in #293

Can you mention what other mutation types are not going well? I believe all cases are covered now.

**Comment by SuperVK** [comment]
Is it possible to open the mutation modal of an invoice now? Because the code doesn't change that but it was broken in production as mentioned in the issue. (I cant test atm).

File: apps/dashboard/src/components/mutations/mutationmodal/ModalDetailPayoutRequest.vue (hunk 1 above)
Comment (CodeNamedRobin) [reply apps/dashboard/src/components/mutations/mutationmodal/ModalDetailPayoutRequest.vue]: Fixed in latest commit.

**Comment by CodeNamedRobin** [comment]
Now fixes #293 fully with the latest commit.

File: apps/dashboard/src/components/mutations/MutationsBalance.vue [hunk 2]
```diff
@@ -132,6 +132,7 @@ const openedMutationType = ref<FinancialMutationType>();
 const isModalVisible = ref<boolean>(false);
 
 function openModal(id: number, type: FinancialMutationType) {
+    console.error(id, type);
```
Comment (SuperVK) [inline apps/dashboard/src/components/mutations/MutationsBalance.vue]: Beep boop

File: apps/dashboard/src/components/mutations/mutationmodal/ModalMutation.vue [hunk 3]
```diff
@@ -115,7 +117,13 @@ const shouldShowWaivedFine = computed(() => {
     props.type === FinancialMutationType.WAIVED_FINE);
 });
 
+const shouldShowPayoutRequest = computed(() => {
+  if (!transferDetails.value[props.id]) return false;
+  return (
+    props.type === FinancialMutationType.PAYOUT_REQUEST);
+});
 async function fetchTransferInfo() {
+  console.error("Fetching transfer info");
```
Comment (SuperVK) [inline apps/dashboard/src/components/mutations/mutationmodal/ModalMutation.vue]: Beep boop 2

**Review by SuperVK — APPROVED** [review]

File: apps/dashboard/src/components/mutations/MutationsBalance.vue (hunk 2 above)
Comment (CodeNamedRobin) [reply apps/dashboard/src/components/mutations/MutationsBalance.vue]: Fixed in latest commit.

File: apps/dashboard/src/components/mutations/mutationmodal/ModalMutation.vue (hunk 3 above)
Comment (CodeNamedRobin) [reply apps/dashboard/src/components/mutations/mutationmodal/ModalMutation.vue]: Fixed in latest commit.

**Pull request merged**